### PR TITLE
Add docstrings, rework reference_fields configuration syntax

### PR DESCRIPTION
**Pull request by chaosk**

### CODE REVIEW

File: djangocms_references/helpers.py [hunk 1]
```diff
@@ -92,6 +158,12 @@ def contenttype_values_queryset(queryset):
 
 
 def convert_plugin_querysets_to_sources(querysets):
+    """Convert provided plugin querysets to a form containing
```
Comment (monikasulik) [inline djangocms_references/helpers.py]: I found this a bit confusing, when I read "form" first time I thought you meant a django form class (which would have been weird!).

File: djangocms_references/helpers.py [hunk 2]
```diff
@@ -92,6 +158,12 @@ def contenttype_values_queryset(queryset):
...
+    only source object information and concatenate them into one
+    using UNION.
```
Comment (monikasulik) [inline djangocms_references/helpers.py]: I'm still confused about what this returns exactly. "source objects" could potentially be anything and I don't think you could really combine any type of queryset using union? So are you returning a CMSPlugin queryset?

File: djangocms_references/cms_config.py [hunk 3]
```diff
@@ -21,19 +22,34 @@ def __init__(self):
...
+        """
         # generate reference_models and reference_plugins dict object
-        for field in fields:
-            model = field.field.model
-            related_model = field.field.related_model
+        for definition in definitions:
+            try:
+                model, field_name = definition
+            except (ValueError, TypeError) as e:
+                raise ImproperlyConfigured(
```
[the remark this answers is not in the document]
Comment (chaosk) [reply djangocms_references/cms_config.py]: There's a test where we pass a list of ints there.

[the remark this answers is not in the document]
Comment (monikasulik) [reply djangocms_references/cms_config.py]: Where is it? :eyeglasses: :older_woman:

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    # project/urls.py
+    urlpatterns += [
+        url(r'', include('djangocms_references.urls'))
+    ]
+
+Configuration
+-------------
+
+References uses Django CMS app registration system.
```
Comment (monikasulik) [inline docs/index.rst]: Missing "the" as in `the Django CMS...`

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    # polls/cms_config.py
+    from cms.app_base import CMSAppConfig
+
+    class PollsAppConfig(CMSAppConfig):
+        djangocms_references_enabled = True
+        reference_fields = [
+            (Answer, 'poll'),
+        ]
+
+In this example, ``Answer`` can contain a reference to ``Poll`` object.
```
Comment (monikasulik) [inline docs/index.rst]: **the** Poll object

**Pull request merged**